src/app/middlewares/access/person.js

```diff
@@ -0,0 +1,21 @@
+const { role } = require('../../utils/enumUser')
+
+module.exports = (action, params, currentRole) => {
+  const { id } = params
+  let permission = false
+
+  switch (action) {
```

Comment:
Cara, é melhor você utilizar o jwt token para controlar isso do que ficar validando rota, você só precisa adicionar esse middleware nas rotas e ele já vai barrar caso não for um admin

Reply:
Eu estou usando JWT, este é um middleware que eu coloquei na rota. Depois do middleware de autenticação que valida o token, ele passa por este para validar o acesso a um recurso específico pelo tipo do usuário.

Reply:
Então, usa o payload que você tem no token. melhor que fazer isso, sempre que criar uma rota nova precisará adicionar no middleware, sendo que se validasse somente o tipo de usuário seria mais fácil

Reply:
Mas eu não posso bloquear todas as rotas apenas para admin, preciso ter o usuário autenticado.

Reply:
Alinhado pelo chat